lib/galaxy/webapps/galaxy/api/authenticate.py

```diff
@@ -92,7 +92,13 @@ def _decode_baseauth(self, encoded_str):
 
         :raises: HTTPBadRequest
         """
-        split = encoded_str.strip().split(' ')
+        try:
+            split = encoded_str.strip().split(' ')
+        except AttributeError:
+            if not encoded_str:
+                raise exceptions.MessageException('Authentication is missing')
+            else:
```

Comment:
In what other circumstances would this throw an AttributeError?

Reply:
if it's called from another function on backend? I thought it won't be good to always return 'Authentication is missing', but I guess you're right. I will remove it.

Reply:
Yeah, what I was getting at was how else would this throw an AttributeError other than encoded_str being missing.  Agreed, the single "RequestParameterInvalidException" is enough.